lib/kafka-worker.rb

```diff
@@ -38,16 +48,20 @@ def run
           if message.topic == handler.topic
             handler_obj = handler.new
             handler_obj.logger = @logger
-            begin
-              handler_obj.handle(message)
-            rescue Exception => err
-              handler_obj.on_error(message, err)
+            (0..3).each do
+              begin
+                handler_obj.handle(message)
+                break
```

Comment:
I was thinking about that as well. Another option is to use a producer and push it back to kafka, but that assumes that the message has arrived properly.